pkgs/development/compilers/opensmalltalk-vm/default.nix

```diff
@@ -103,84 +92,102 @@ let
       '';
 
       meta = {
-        description = "The cross-platform virtual machine for Squeak, Pharo, Cuis, and Newspeak.";
-        mainProgram = scriptName;
+        description =
+          "The cross-platform virtual machine for Squeak, Pharo, Cuis, and Newspeak.";
+        mainProgram = finalAttrs.scriptName;
         homepage = "https://opensmalltalk.org/";
         license = with lib.licenses; [ mit ];
-        maintainers = with lib.maintainers; [ jakewaksbaum ];
-        platforms = [ stdenv.targetPlatform.system ];
+        maintainers = with lib.maintainers; [ jakewaksbaum ehmry ];
+        inherit platforms;
       };
     };
 
-  vmsByPlatform = {
-    "aarch64-linux" = {
-      "squeak-cog-spur" = buildVM {
+  buildVM = { vmName, platforms }:
+    stdenv.mkDerivation (final:
+      let
+        pkgFunc = platforms.${stdenv.targetPlatform.system} or (_: _: { });
+        prev = baseFunc {
+          inherit vmName;
+          platforms = builtins.attrNames platforms;
+        } final;
+      in prev // pkgFunc final prev);
```

Comment:
Here `mkDerivation` is being called with a composed function rather than use `overrideAttrs` later. Using `overrideAttrs` still works, but works better than before because things like `vmName`, `scriptName`, and `platformDir` can be overriden later.